src/containers/Application/ApplicationCreate.tsx

```diff
@@ -3,42 +3,47 @@ import { TemplateTypePayload } from '../../utils/types'
 import { ApplicationStart, Loading } from '../../components'
 import { useApplicationState } from '../../contexts/ApplicationState'
 import { useRouter } from '../../utils/hooks/useRouter'
-import { Header } from 'semantic-ui-react'
+import { Header, Message } from 'semantic-ui-react'
 import useLoadTemplate from '../../utils/hooks/useLoadTemplate'
-import useCreateApplication from '../../utils/hooks/useCreateApplication'
+import useCreateApplication, {
+  createApplicationProps,
+} from '../../utils/hooks/useCreateApplication'
 
 const ApplicationCreate: React.FC = () => {
   const { applicationState, setApplicationState } = useApplicationState()
   const { serialNumber } = applicationState
   const { push, query } = useRouter()
   const { type } = query
 
-  const { applicationMutation, responses } = useCreateApplication()
+  const { processing, error: creationError, create } = useCreateApplication()
 
-  const { apolloError, error, loading, templateType, templateSections } = useLoadTemplate({
+  const {
+    apolloError,
+    error,
+    loading,
+    templateType,
+    templateSections,
+    templateElementsIds,
+  } = useLoadTemplate({
     templateCode: type as string,
   })
 
   useEffect(() => {
-    if (
-      Object.values(responses).length > 0 && // Checking the responses are created
-      !Object.values(responses).some((value) => value === false) &&
-      serialNumber &&
-      templateSections &&
-      templateSections.length > 0
-    ) {
+    if (!processing && serialNumber && templateSections && templateSections.length > 0) {
       // Call Application page on first section
       const firstSection = templateSections[0].code
       // The pageNumber starts in 1 when is a new application.
       const pageNumber = 1
       push(`${serialNumber}/${firstSection}/page${pageNumber}`)
```

Comment:
I think this push navigation should happen in the callback of create mutation

Reply:
Oh, nice! That's a good idea I'll see if I can change to that now.

Reply:
Done